R/generate_pt_nowcast_mat_list.R

```diff
@@ -64,17 +64,48 @@ generate_pt_nowcast_mat_list <- function(reporting_triangle_list,
   } else { # create a list with the same pmf
     delay_pmf_list <- rep(list(delay_pmf), length(reporting_triangle_list))
   }
-  # Iterate through each item in both lists of reporting triangles
-  # and delay PMFs
+
+  safe_generate_pt_nowcast_mat <- .safelydoesit(generate_pt_nowcast_mat)
+
+  # Use the safe version in mapply, iterating through each item in both
+  # lists of reporting triangles and delay PMFs
   pt_nowcast_mat_list <- mapply(
-    generate_pt_nowcast_mat,
+    function(triangle, pmf) {
+      result <- safe_generate_pt_nowcast_mat(
+        reporting_triangle = triangle,
+        delay_pmf = pmf,
+        n = n,
+        max_delay = max_delay
+      )
+      if (!is.null(result$error)) {
+        # Return NULL if there was an error
```

Comment:
here I suggest returning the errors as a warning or message along with the index lof the list. To me this suggests reworking this iterator to work other an index vs the current approach.

Reply:
Are you suggesting that you want the result of this function to be a nested list, where each indexed element contains a `result` (either NULL or a matrix) and an `error` (either NULL or the nested list of error information)?

Reply:
no I am suggesting printing them as a warning or message. take a look at the scoringutils iterator

Reply:
I have updated this to provide a message with the indices and the error message corresponding to the call to `generate_pt_nowcast()`